test/performance/benchmarks_test.go

```diff
@@ -85,41 +86,31 @@ func runTest(t *testing.T, img string, baseQPS float64, loadFactors []float64) {
 		t.Fatalf("Error probing domain %s: %v", domain, err)
 	}
 
-	opts := loadgenerator.GeneratorOptions{
-		Duration: duration,
-		// Use the same number of threads as the QPS to send the requests
-		NumThreads: int(baseQPS),
-		// Use the same number of connections as the QPS to avoid TIME_OUT state sockets, see
-		// http://tleyden.github.io/blog/2016/11/21/tuning-the-go-http-client-library-for-load-testing/
-		NumConnections: int(baseQPS),
-		Domain:         domain,
-		URL:            fmt.Sprintf("http://%s", *endpoint),
-		RequestTimeout: reqTimeout,
-		BaseQPS:        baseQPS,
-		LoadFactors:    loadFactors,
-		FileNamePrefix: filename(tName),
-	}
-	resp, err := opts.RunLoadTest(loadgenerator.AddHostHeader)
-	if err != nil {
-		t.Fatalf("Generating traffic via fortio failed: %v", err)
-	}
+	targeter := vegeta.NewStaticTargeter(vegeta.Target{
+		Method: "GET",
+		URL:    fmt.Sprintf("http://%s", *endpoint),
+	})
+	attacker := vegeta.NewAttacker()
 
-	// Save the json result for benchmarking
-	resp.SaveJSON()
+	var metrics vegeta.Metrics
+	for res := range attacker.Attack(targeter, pacer, duration, tName) {
+		metrics.Add(res)
+	}
+	metrics.Close()
 
-	if len(resp.Result) == 0 {
-		t.Fatal("No result found for the load test")
+	// Return directly if we do not want to save metrics for this test run
+	if !saveMetrics {
+		return
 	}
 
 	var tc []junit.TestCase
 	// Add latency metrics
-	for _, p := range resp.Result[0].DurationHistogram.Percentiles {
-		val := float32(p.Value) * 1000
-		name := fmt.Sprintf("p%d(ms)", int(p.Percentile))
-		tc = append(tc, perf.CreatePerfTestCase(val, name, tName))
-	}
+	tc = append(tc, perf.CreatePerfTestCase(float32(metrics.Latencies.P50)*1000, "p50(ms)", tName))
```

Comment:
You can probablyse use `..P50.Seconds()*1000`.

Reply:
Done.